responsibleai/responsibleai/_tools/shared/state_directory_management.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+"""Directory management of state of ResponsibleAI insights."""
+
+import os
+import uuid
+from pathlib import Path
+
+
+class DirectoryManager:
+
+    CONFIGURATION = 'configuration'
+    DATA = 'data'
+    EXPLAINER = 'explainer'
+
+    def __init__(self, parent_directory_path, sub_directory_name=None):
+        """Directory manager for managing the configuration,
+           dashboard data and explainers need to recompose the state
+           in the various managers.
+
+        :param parent_directory_path: Directory path of where the state data
+                                      directory will be stored.
+        :type parent_directory_path: Path
+        :param sub_directory_name: The name where the state needs to be stored
+                                   or the directory where the state can be
+                                   found.
+        :type sub_directory_name: str
+        """
+        self.parent_directory_path = parent_directory_path
+
+        if sub_directory_name is None:
+            self.sub_directory_name = str(uuid.uuid4())
+        else:
+            self.sub_directory_name = sub_directory_name
+
+        parent_directory = Path(self.parent_directory_path)
+        if not parent_directory.exists():
+            parent_directory.mkdir(parents=True, exist_ok=True)
+
+        sub_directory = (Path(self.parent_directory_path) /
+                         self.sub_directory_name)
+        if not sub_directory.exists():
+            sub_directory.mkdir(parents=True, exist_ok=True)
+
+    def create_config_directory(self):
+        """Create the configuration directory and return its path.
+
+        :returns: Path to the configuration directory.
+        """
+        config_directory = (Path(self.parent_directory_path) /
+                            self.sub_directory_name /
+                            DirectoryManager.CONFIGURATION)
+        if not config_directory.exists():
+            config_directory.mkdir(parents=True, exist_ok=True)
+
+        return config_directory
+
+    def create_data_directory(self):
+        """Create the dashboard data directory and return its path.
+
+        :returns: Path to the dashboard data directory.
+        """
+        data_directory = (Path(self.parent_directory_path) /
+                          self.sub_directory_name /
+                          DirectoryManager.DATA)
+        if not data_directory.exists():
+            data_directory.mkdir(parents=True, exist_ok=True)
+
+        return data_directory
+
+    def create_explainer_directory(self):
+        """Create the explainer directory and return its path.
```

Comment:
this seems something too specific to explanations to be in this class?  Just seems a little weird to put it in a more common RAI class.  I wonder if there is a better way to structure the DirectoryManager.

Reply:
Will this also be used for causal model in the future?  Maybe the name could just be made more generic somehow.

Reply:
It looks like it's not used currently by the managers if I understand the code correctly right now.

Reply:
How about this: we have different directory managers for different components, and the explanation and counterfactual components can use the ExplainerDirectoryManager which includes this explainer directory.

Reply:
As per the design dicussion we decided on maintaining the same directory structure for all the components. Do you have any name suggestion for explainer_path? 

For causal manager, I will move the causal models to explainer path in a follow up PR. The state management for causal is alittle bit tricky.